src/ResourceManager/EventGrid/Commands.EventGrid/EventSubscription/UpdateAzureEventGridSubscription.cs

```diff
@@ -0,0 +1,316 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.EventGrid.Models;
+using Microsoft.Azure.Commands.EventGrid.Utilities;
+using Microsoft.Azure.Management.EventGrid.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.EventGrid
+{
+    [Cmdlet(
+        VerbsData.Update,
+        EventGridEventSubscriptionVerb,
+        SupportsShouldProcess = true,
+        DefaultParameterSetName = ResourceGroupNameParameterSet),
+     OutputType(typeof(PSEventSubscription))]
+    public class UpdateAzureEventGridSubscription : AzureEventGridCmdletBase
+    {
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "The identifier of the resource to which the event subscription was created.",
+            ParameterSetName = ResourceIdEventSubscriptionParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipeline = true,
+            Position = 0,
+            HelpMessage = EventGridConstants.EventSubscriptionInputObject,
+            ParameterSetName = EventSubscriptionInputObjectParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public PSEventSubscription InputObject { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
```

Comment:
@kalyanaj you can remove the `ValueFromPipelineByPropertyName` property from all parameters that aren't `ResourceId`

Reply:
OK, done. That said, I have this pattern in a few other commands too: for example, in the below case, the resourcegroup name and topicname parameters are picked up from the pipeline properties. Is it not required to support such cases?

Get-AzureRmEventGridTopic -ResourceGroup $resourceGroupName -TopicName $topicName | Remove-AzureRmEventGridSubscription -EventSubscriptionName $eventSubscriptionName

Reply:
@kalyanaj the two piping scenarios we require to support are

(1) Piping an object by value (_e.g._, `Get-AzureRmEventGridSubscription | Remove-AzureRmEventGridSubscription`)
(2) Piping the generic resource object so the `ResourceId` gets picked up by property name (_e.g._, `Find-AzureRmResource -ResourceType Microsoft.EventGrid/eventSubscriptions | Remove-AzureRmEventGridSubscription`)

In this case, since you're able to pass objects across different resource cmdlets, it might make sense to continue to support `ValueFromPipelineByPropertyName` for these additional parameters as you see fit.